src/parser/core/modules/CooldownDowntime.js

```diff
@@ -43,27 +43,30 @@ export default class CooldownDowntime extends Module {
 		const encounterLength = endTime - startTime
 		const OGCDRequirements = []
 
-		this.trackedCds.map(id => {
-			const allowedDowntime = Number.isInteger(this.allowedDowntimePerOgcd[id]) ? this.allowedDowntimePerOgcd[id] : this.allowedDowntime
-			const firstUseOffset = Number.isInteger(this.firstUseOffsetPerOgcd[id]) ? this.firstUseOffsetPerOgcd[id] : this.firstUseOffset
-			const timeOnCooldown = this.cooldowns.getTimeOnCooldown(id, true, allowedDowntime)
-
-			//calculate the downtime based on the start and stop values and sum the array
-			//Adjust for the classes defined alloted time to allow a CD to be held
-			//this supports classes like RDMs who routinely hold CDs due to procs
-			//write the results as a new Requirement to show up later
-			OGCDRequirements.push(
-				new Requirement({
-					name: <ActionLink {...getDataBy(ACTIONS, 'id', id)} />,
-					percent: this._percentFunction(
-						id,
-						encounterLength - timeOnCooldown + allowedDowntime,
-						firstUseOffset,
-						encounterLength
-					),
-				})
-			)
+		const groups = {}
+		this.trackedCds.forEach(id  => {
+			const groupId = getDataBy(ACTIONS, 'id', id).cooldownGroup || 0
```

Comment:
This has the theoretical potential to error, if the ID is ever not found. Probably _ok_ in this situation, but avoidable.

I'd rather use something like `'unknown'` or `'ungrouped'` for the fallback, though - `0` is a bit of a magic value.